pythonpro/modules/templates/modules/module_index.html

```diff
@@ -22,11 +22,41 @@ <h5 class="card-title">Objetivo</h5>
                             <a href="{% url 'modules:description' slug=module.slug %}" class="btn btn-secondary">Saiba
                                 Mais</a>
                             {% if module.title == 'Python Birds' %}
-                                <a href="{% url 'core:lead_landing' %}" class="btn btn-success"> Eu quero!</a>
-                            {% elif module.title == 'PyTools' or module.title == 'Django' %}
-                                <a href="{% url 'checkout:webdev_landing_page' %}" class="btn btn-success"> Comprar</a>
+                                {% if user|has_role:'lead,client,webdev,bootcamper,pythonista,member' %}
+                                    <a href="{{ module.get_absolute_url }}" class="btn btn-success"> Acessar</a>
```

Comment:
Code could be simpler here if you create 2 dynamic properties Module´s on view: has_access and buy_link.

has_access you can generate using this function: https://github.com/pythonprobr/pythonpro-website/blob/master/pythonpro/modules/topics_views.py#L27.

For buy link you can generate a dict conectin module´s slug to it´s buy link.

Reply:
it's done